.changeset/new-toes-brake.md

```diff
@@ -0,0 +1,6 @@
+---
+"@hashicorp/design-system-components": patch
```

Comment:
Since this is technically a (potentially) breaking change(de facto, we're changing the API of the component, its intrinsic behaviour) should we raise it at least to `minor`?

Reply:
It doesn't seem like it would break current usage. Can you think of any scenarios that could potentially cause issues that could be tested though?

Reply:
The only breaking scenario I can think of is if someone invoked the subcomponents in a different order than the suggested one.

The example below would currently be corrected by the component and rendered in the pre-defined order (Header, Body, Footer).
```
  <Hds::Modal id="basic-modal" @onClose={{fn this.deactivateModal "basicModalActive"}} as |M|>
    <M.Body>
      <p class="hds-typography-body-300 hds-foreground-primary">Modal content</p>
    </M.Body>
    <M.Header>Modal title</M.Header>
    <M.Footer as |F|>
      <Hds::Button type="button" @text="Confirm" {{on "click" F.close}} />
    </M.Footer>
  </Hds::Modal>
```

After this change, the subcomponents would render in the order they were invoked in (Body, Header, Footer).

Reply:
I think there's an incredibly low risk of these being invoked in the "wrong" order. Header > Body > Footer is overwhelmingly standard.